### DISCUSSION TITLE
当编译出错的时候，如何快速定位错误位置？

### DISCUSSION THREAD

**Discussion by bakulaibuji** [sjtug/SJTUThesis#860] [提问]
相信很多小伙伴在latex编写的时候都遇到相似的问题，可能会一直编辑编辑再编译。如果这个时候编译出现了问题，如何快速定位出问题位置呢？(main.log太长，怎么样快速在main.log中定位问题呢)

**Comment by LogCreative** [comment]
- 如果使用的是 Overleaf，会在报错对应的行号标注错误。
- 如果是本地编译，可以考虑使用专用编辑器与对应插件，详见 [wiki](https://github.com/sjtug/SJTUThesis/wiki/本地使用说明#使用编辑器进行编译)。
- 命令行也可以查看对应的错误，见 [处理错误](https://github.com/CTeX-org/learnlatex.github.io/blob/zh-Hans/zh-Hans/lesson-15.md) 和 [更多：处理错误](https://github.com/CTeX-org/learnlatex.github.io/blob/zh-Hans/zh-Hans/more-15.md)。

**Reply by bakulaibuji** [reply]
我已经在vscode中按照要求设置了，但是报错了之后仍然需要查看main.log去处理错误，所以你提到的插件有便捷查看错误的功能吗

**Reply by LogCreative** [reply]
考虑使用 vscode 插件自带的 recipe 编译，可以接入 vscode 本身的 lint 进行分析，也可以配合 synctex 进行跳转。SJTUThesis 可以直接使用 latexmk(xelatex) 这个默认的 recipe。

详见 [LaTeX-Workshop wiki](https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#catching-errors-and-warnings)。

**Reply by bakulaibuji** [reply]
我在接入vscode的output时发现，vscode做的就是把main.log打印到了vscode output里，你说的vscode的lint是可以对main.log进行分析吗？可以简单介绍下怎么做吗？

**Reply by LogCreative** [reply]
理论上使用 LaTeX Workshop 插件编译应该会把报错信息显示在 问题（Problems） 选项卡，并在文件中标记红色。只使用 终端（Terminal） 手动输入编译指令是不行的。

**Reply by bakulaibuji** [reply]
现在清楚了，十分感谢
